bot/university/university.py

```diff
@@ -128,6 +143,293 @@ def check_reaction(_reaction, user):
+                                        course_channel: discord.TextChannel,
+                                        offered_group: int):
+    """Builds an embed with information about new group exchange offers.
+
+    Args:
+        author (discord.User): The author of the new offer.
+        message (discord.Message): The message containing the new offer.
+        course_channel (discord.TextChannel): The channel referring to the course of the offer.
+        offered_group (int): The group that is offered.
+
+    Returns:
+        (discord.Embed): An embed containing all information above for the user.
+    """
+    course_name = _parse_course_from_channel_name(course_channel)
+    return discord.Embed(title="Wir haben ein neues Tauschangebot für dich gefunden:",
+                         description="Bitte vergiss nicht, deine Anfrage mit dem Befehl 'exchange remove "
+                                     "<channel-mention>' wieder zu löschen, wenn du einen Tauschpartner gefunden "
+                                     "hast",
+                         color=constants.EMBED_COLOR_GROUP_EXCHANGE) \
+        .set_thumbnail(url=author.avatar_url) \
+        .add_field(name="User:", value="[{0}]({1})".format(author, message.jump_url)) \
+        .add_field(name="Kurs:", value=course_name) \
+        .add_field(name="Bietet:", value="Gruppe {0}".format(offered_group))
+
+
+def _build_group_exchange_embed(author: discord.User,
+                                channel: discord.TextChannel,
+                                offered_group: int,
+                                requested_groups: Iterable[int]) -> discord.Embed:
+    """Builds an embed for a group exchange request.
+
+    The embed contains information about who wants to exchange groups, which groups he offers and requests and for what
+    courses.
+
+    Args:
+        author (discord.User): The author of the embed (aka the user who wants to change groups).
+        channel (discord.TextChannel): The channel that refers to the course that should be changed.
+        offered_group (int): The group that the user offers.
+        requested_groups (List[int]): The groups that the user would accept.
+
+    Returns:
+        (discord.Embed): The embed with the group exchange request.
+    """
+    return discord.Embed(title=_parse_course_from_channel_name(channel),
+                         color=constants.EMBED_COLOR_GROUP_EXCHANGE) \
+        .set_thumbnail(url=author.avatar_url) \
+        .add_field(name="Biete:",
+                   value="Gruppe {0}".format(offered_group)) \
+        .add_field(name="Suche:",
+                   value="Gruppen {0}".format(", ".join(map(str, requested_groups)))) \
+        .add_field(name="Eingereicht von:",
+                   value="{0}\n{0.mention}".format(author),
+                   inline=False)
+
+
+def _parse_course_from_channel_name(channel: discord.TextChannel):
+    """Parses the course name from a discord channel name
```

Comment:
Changed to:

```
Parses the course name from a discord text channel.

The channel name must contain at least one channel name separator emoji (as specified in constants.py) and it will
reformat the name to replace all dashes with spaces and capitalize every word.
```

bc. it does not have to be a channel regarding a course, but we use it for that. I could use it for other purposes where this format is part. Makes the documentation a bit more robust if we ever change something and forget the docu (what will inavoidably happen :D)